### DISCUSSION TITLE
regex-fied domain

### DISCUSSION THREAD

**Discussion by Yuki2718** [uBlockOrigin/uBlock-issues#2234] [Ideas] [answered]
(transferred from internal discussion https://github.com/orgs/uBlockOrigin/teams/ublock-filters-volunteers/discussions/423 so some details are outdated)

krystian3w notified me that AG is implementing this: https://github.com/AdguardTeam/CoreLibs/issues/1550

I think this will be game changer, for example we'll be able to address those [pixsera crap](https://github.com/uBlockOrigin/uAssets/search?q=pixsera&type=commits) like
```
@@*$ghide,domain=/img[a-z]{3,4}\.buzz/
/img[a-z]{3,4}\.buzz/##+js(aeld, , _0x)
```
The format is just my example idea, AG seems to be going with the same format as $path but we don't have to follow. Although I expect pixsera will then fully randomize domain, the regex will also be useful to other cases such as xhamster proxy sites or some pirate sites.

**Comment by gorhill** [comment, marked as answer]
>     @@*$ghide,domain=/img[a-z]{3,4}\.buzz/

https://github.com/gorhill/uBlock/commit/b1de8d3fe48755da58268ba86dccd5d76940b613

**Reply by gorhill** [reply]
>     /img[a-z]{3,5}\.buzz/##+js(nowoif)

https://github.com/gorhill/uBlock/commit/81498474d6d440b032681aa9952d593749b39efb

**Reply by Yuki2718** [reply]
Ty for all your time and continuous effort!

**Reply by v1gnesh** [reply]
Sincere thanks for everyone's work on this... seriously!

I dream for a day when I can automatically convert my long list of dynamic filtering rules to static filtering rules.
And on that day, when all rules are static filters, baking [uBO-core](https://www.npmjs.com/package/@gorhill/ubo-core) into some network-wide filtering web proxy will be very close to achievable.

**Reply by v1gnesh** [reply]
Why confused, @krystian3w ?

**Comment by uBlock-user** [comment]


**Reply by krystian3w** [reply]
works both:

```adb

! "negation" only com
/example\.(?!com)/##body > *

! "negation" normal 'regex'
~/example\.com/##body > * 
```

**Reply by uBlock-user** [reply]
No, I meant `~/www.google.com/,/com/##+js(nowoif)` and yes it works as expected.

**Comment by MasterKia** [comment]
Added to wiki:

https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#domain
https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#to

https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#general-purpose-scriptlets (Last point)

**Reply by uBlock-user** [reply]
> This new syntax does not apply to static extended filters.

Remove this line, support has been added - https://github.com/gorhill/uBlock/commit/81498474d6d440b032681aa9952d593749b39efb

**Reply by MasterKia** [reply]
Done.

**Reply by krystian3w** [reply]
See these maybe regex allow change from https://github.com/uBlockOrigin/uBlock-issues/labels/declined to https://github.com/uBlockOrigin/uBlock-issues/labels/fixed: https://github.com/uBlockOrigin/uBlock-issues/issues/957 / fyi  @gwarser

**Reply by ghost** [reply]
> in uBlock's My Rules panel.

He asked for rule syntax.

**Reply by krystian3w** [reply]
But is "inactive" after 11 january 2023.

**Comment by krystian3w** [comment]
I guess "negated" regex don't allow `,`. https://github.com/uBlockOrigin/uBlock-issues/discussions/2234#discussioncomment-4833698

```adb
example.*,~/example\.([a-z]{1,2}|[a-z]{4,16})/##body > *
```

![image](https://user-images.githubusercontent.com/35370833/227167418-40d98400-c6ac-4489-b323-5e6f2e6a9556.png)

Workaround can be:
```adb
example.*##body > *
/example\.([a-z]{1,2}|[a-z]{4,16})/#@#body > *
```

by virtue of the fact that the parser does not have a noop state for cosmetic filters like other silinics.

---

and maybe `|`:

```adb
$all,~doc,domain=example.*|~/example\.([a-z]{1,2}|[a-z]{4,16})/
```

![image](https://user-images.githubusercontent.com/35370833/227168199-b59072d4-9b6e-43cd-ae6d-e1c058ac89ca.png)

Lack of workaround to avoid mark allowed to load and block latter with `,important` modifier, I doubt that it can be used with `,bafilter`, `denyallow` or `to=`.

**Comment by stephenhawk8054** [comment]
Currently when I use `to=~/youtube/`, the filter automatically translates to ``to=~~/youtube/`` and cannot exclude domain with `youtube` inside.

Steps to reproduce:

1. Add

```adb
*$frame,from=plainlight.com,to=~/youtube/
```

2. Access `https://www.plainlight.com/how-to-embed-youtube-videos.html`
3. The filter becomes

```adb
*$frame,from=plainlight.com,to=~~/youtube/
```

and cannot exclude `youtube-nocookie.com` domain: https://user-images.githubusercontent.com/66517106/232954843-e6dc927b-0e0c-47be-8cf6-b6f20d2a6ce7.png

---

If I use

```adb
*$frame,from=plainlight.com,to=/youtube/
```

then it blocks the `youtube` domain successfully as expected.: https://user-images.githubusercontent.com/66517106/232955214-c399a4e2-2478-4cb6-ab67-6185a3a6c38c.png

**Reply by gorhill** [reply]
Fixed with <https://github.com/gorhill/uBlock/commit/3dddf4ca47bac1b165794575d4e6eb1f24d3deea>. Unfortunate that I published stable release yesterday before this was found.

**Reply by stephenhawk8054** [reply]
Confirm it's fixed now:

![image](https://user-images.githubusercontent.com/66517106/233067357-7715f096-9501-4cbe-96af-c17cf1e04132.png)
